src/inventory/mod.rs

```diff
@@ -252,7 +411,15 @@ pub struct Item {
     pub material: Material,
 }
 
-#[derive(Debug, Clone)]
+impl Item {
+    /// Check if this item stack matches another, allowing these to be merged
+    /// on an inventory slot.
+    pub fn matches(&self, other: &Item) -> bool {
```

Comment:
The description is pretty decent here, but the function name doesn't really tell anything about the stacking requirements, so `is_stackable` would probably a more descriptive name, also I am not sure but should we maybe turn this into a `PartialEq` impl? This might be undesirable though because this method doesn't check full equality, so not sure about that one